diff --git a/sound/soc/mediatek/mt8196/mt8196-afe-common.h b/sound/soc/mediatek/mt8196/mt8196-afe-common.h
index ae3fcfc8a88e895fdc111e605d115518a5d41ce9..8cbdf1d6895a189495ad91f675f95a6c549f42c7 100644
--- a/sound/soc/mediatek/mt8196/mt8196-afe-common.h
+++ b/sound/soc/mediatek/mt8196/mt8196-afe-common.h
@@ -272,6 +272,7 @@ struct mt8196_afe_private {
 
 	/* channel merge */
 	unsigned int cm_rate[CM_NUM];
+	unsigned int cm_channels;
 };
 
 int mt8196_dai_adda_register(struct mtk_base_afe *afe);
diff --git a/sound/soc/mediatek/mt8196/mt8196-afe-pcm.c b/sound/soc/mediatek/mt8196/mt8196-afe-pcm.c
index edd208dfac9e8ce4cea4c7accc2e5dd1204f8525..d94116da4b483047ff712785610d725eb9327f4c 100644
--- a/sound/soc/mediatek/mt8196/mt8196-afe-pcm.c
+++ b/sound/soc/mediatek/mt8196/mt8196-afe-pcm.c
@@ -129,6 +129,19 @@ void mt8196_fe_shutdown(struct snd_pcm_substream *substream,
 	}
 }
 
+int mt8196_fe_hw_params(struct snd_pcm_substream *substream,
+			 struct snd_pcm_hw_params *params,
+			 struct snd_soc_dai *dai)
+{
+	struct mtk_base_afe *afe = snd_soc_dai_get_drvdata(dai);
+	unsigned int channels = params_channels(params);
+	struct mt8196_afe_private *afe_priv = afe->platform_priv;
+
+	afe_priv->cm_channels = channels;
+
+	return mtk_afe_fe_hw_params(substream, params, dai);
+}
+
 int mt8196_fe_trigger(struct snd_pcm_substream *substream, int cmd,
 				struct snd_soc_dai *dai)
 {
@@ -298,7 +311,7 @@ int mt8196_get_memif_pbuf_size(struct snd_pcm_substream *substream)
 static const struct snd_soc_dai_ops mt8196_memif_dai_ops = {
 	.startup        = mt8196_fe_startup,
 	.shutdown       = mt8196_fe_shutdown,
-	.hw_params      = mtk_afe_fe_hw_params,
+	.hw_params      = mt8196_fe_hw_params,
 	.hw_free        = mtk_afe_fe_hw_free,
 	.prepare        = mtk_afe_fe_prepare,
 	.trigger        = mt8196_fe_trigger,
@@ -816,14 +829,14 @@ static int ul_cm0_event(struct snd_soc_dapm_widget *w,
 {
 	struct snd_soc_component *cmpnt = snd_soc_dapm_to_component(w->dapm);
 	struct mtk_base_afe *afe = snd_soc_component_get_drvdata(cmpnt);
-	unsigned int channels = 0;
+	struct mt8196_afe_private *afe_priv = afe->platform_priv;
+	unsigned int channels = afe_priv->cm_channels;
 
-	dev_dbg(afe->dev, "%s(), event 0x%x, name %s\n",
-		 __func__, event, w->name);
+	dev_info(afe->dev, "%s(), event 0x%x, name %s, channels %d\n",
+		 __func__, event, w->name, channels);
 
 	switch (event) {
 	case SND_SOC_DAPM_PRE_PMU:
-		channels = 3;
 		mt8196_enable_cm_bypass(afe, CM0, 0x0);
 		mt8196_set_cm(afe, CM0, true, false, channels);
 		break;
@@ -842,14 +855,14 @@ static int ul_cm1_event(struct snd_soc_dapm_widget *w,
 {
 	struct snd_soc_component *cmpnt = snd_soc_dapm_to_component(w->dapm);
 	struct mtk_base_afe *afe = snd_soc_component_get_drvdata(cmpnt);
-	unsigned int channels = 0;
+	struct mt8196_afe_private *afe_priv = afe->platform_priv;
+	unsigned int channels = afe_priv->cm_channels;
 
-	dev_info(afe->dev, "%s(), event 0x%x, name %s\n",
-		 __func__, event, w->name);
+	dev_info(afe->dev, "%s(), event 0x%x, name %s, channels %d\n",
+		 __func__, event, w->name, channels);
 
 	switch (event) {
 	case SND_SOC_DAPM_PRE_PMU:
-		channels = 4;
 		mt8196_enable_cm_bypass(afe, CM1, 0x0);
 		mt8196_set_cm(afe, CM1, true, false, channels);
 		break;
@@ -868,14 +881,14 @@ static int ul_cm2_event(struct snd_soc_dapm_widget *w,
 {
 	struct snd_soc_component *cmpnt = snd_soc_dapm_to_component(w->dapm);
 	struct mtk_base_afe *afe = snd_soc_component_get_drvdata(cmpnt);
-	unsigned int channels = 0;
+	struct mt8196_afe_private *afe_priv = afe->platform_priv;
+	unsigned int channels = afe_priv->cm_channels;
 
-	dev_info(afe->dev, "%s(), event 0x%x, name %s\n",
-		 __func__, event, w->name);
+	dev_info(afe->dev, "%s(), event 0x%x, name %s, channels %d\n",
+		 __func__, event, w->name, channels);
 
 	switch (event) {
 	case SND_SOC_DAPM_PRE_PMU:
-		channels = 4;
 		mt8196_enable_cm_bypass(afe, CM2, 0x0);
 		mt8196_set_cm(afe, CM2, true, false, channels);
 		break;
