From 5ac7baf31bee00908b43b643f24a27264b1cfa03 Mon Sep 17 00:00:00 2001
From: Dexter Rivera <riverade@google.com>
Date: Tue, 3 Dec 2019 18:02:39 -0800
Subject: [PATCH] customize-containerd-service

---
 containerd.service | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/containerd.service b/containerd.service
index 5452ea1f..854a6126 100644
--- a/containerd.service
+++ b/containerd.service
@@ -5,11 +5,13 @@ After=network.target local-fs.target
 
 [Service]
 ExecStartPre=-/sbin/modprobe overlay
-ExecStart=/usr/local/bin/containerd
+ExecStart=/usr/bin/containerd
 
 Delegate=yes
 KillMode=process
+OOMScoreAdjust=-999
 Restart=always
+RestartSec=10
 # Having non-zero Limit*s causes performance problems due to accounting overhead
 # in the kernel. We recommend using cgroups to do container-local accounting.
 LimitNPROC=infinity
-- 
2.24.0.393.g34dc348eaf-goog

