diff --git a/examples/chrony.conf.example1 b/examples/chrony.conf.example1
index 5e93ea7..424a24b 100644
--- a/examples/chrony.conf.example1
+++ b/examples/chrony.conf.example1
@@ -1,5 +1,5 @@
-# Use public NTP servers from the pool.ntp.org project.
-pool pool.ntp.org iburst
+# Use public NTP servers from the metadata.google.internal.
+server metadata.google.internal iburst
 
 # Record the rate at which the system clock gains/losses time.
 driftfile /var/lib/chrony/drift
